docs/source/docs/level-up/intermediate_skills/09_data_filtering.rst

```diff
@@ -0,0 +1,65 @@
+===========================
+Level 9: Dataset Filtering
+===========================
+
+With the increasing availability of public data, the need for data filtering has become more apparent. Raw data often
+contains irrelevant or unnecessary information, making it difficult to extract the desired insights or use it effectively
+for decision-making purposes. Data filtering involves the process of identifying and selecting relevant data points while
+excluding or removing the irrelevant ones to improve the quality and usability of the data. This process is essential for
+ensuring that data can be used effectively and efficiently to drive insights and inform decisions. As the volume and complexity
+of data continue to grow, data filtering will become an increasingly important aspect of data management and analysis.
+By filtering the dataset in this way, we can create a subset of data that is tailored to our specific needs, making it easier
+to extract meaningful insights or use it effectively for decision-making purposes.
+
+In this tutorial, we provide the simple example of filtering dataset using item and annotation. To set how to filter dataset,
+which satisfied some condition, we use XML as query format. Refer this `XPATH <https://devhints.io/xpath>`_ to set your own filter.
+The detailed description for filter operation is given by :doc:`Filter <../../command-reference/context_free/filter>`.
+The more advanced Python example is given :doc:`this notebook <../../jupyter_notebook_examples/notebooks/04_filter>`.
+
+.. tab-set::
+
+    .. tab-item:: ProjectCLI
+
+        With the project-based CLI, we first create project and import datasets into the project
+
+        .. code-block:: bash
+
+            datum project create --output-dir <path/to/project>
+            datum project import --format datumaro --project <path/to/project> <path/to/data>
+
+        We filter dataset through
+
+        .. code-block:: bash
+
+            datum filter -e <how/to/filter/dataset> --project <path/to/project>
+
+        We can set ``<how/to/filter/dataset>`` as your own filter like ``'/item/annotation[label="cat" and area > 85]'``.
+        This example commands will filter only items through the bbox annotations which have `cat` label and bbox area (`w * h`) more than 85.
```

Comment:
```suggestion
        This example command will filter only items through the bbox annotations which have `cat` label and bbox area (`w * h`) more than 85.
```

Reply:
To my knowledge, `'/item/annotation[...]'` removes annotations not items, but your sentence seems removing the items themselves (`'/item[annotation...]'` removes items).

https://github.com/openvinotoolkit/datumaro/blob/8fe4cf0a3106d574968bcfd2e32de1fc576f45bb/datumaro/cli/commands/filter.py#L73-L82

Did you verify the actual behavior of Datumaro for this command? The scope of writing skill up page task contains the verification also.

Reply:
I checked this command myself. Before this command, the dataset had 5000 items, and after this command the dataset result only had 184 items.

Reply:
I closely looked into it and found that `--mode` is an important argument for this problem. This is because it determines whether to filter items or annotations. Therefore, explanation for `--mode` should be added to this section.

https://github.com/openvinotoolkit/datumaro/blob/8fe4cf0a3106d574968bcfd2e32de1fc576f45bb/datumaro/cli/commands/filter.py#L97-L104

https://github.com/openvinotoolkit/datumaro/blob/8fe4cf0a3106d574968bcfd2e32de1fc576f45bb/datumaro/cli/util/project.py#L193-L204

https://github.com/openvinotoolkit/datumaro/blob/8fe4cf0a3106d574968bcfd2e32de1fc576f45bb/datumaro/components/dataset.py#L897-L900

https://github.com/openvinotoolkit/datumaro/blob/8fe4cf0a3106d574968bcfd2e32de1fc576f45bb/datumaro/components/filter.py#L289-L291

What did you give for `--mode`? It seems that you gave `-m i` so that the items disappeared. Then, what is the difference between `'/item/annotation[...]'` and `'/item[annotation...]'`?

Reply:
When I applied the command like `datum filter source-1 -e '/item/annotation[label="cat" and area > 85]'`, `dinfo` for `source-1` is
```
length: 184
categories: label
  label:
    count: 80
    labels: person, bicycle, car, motorcycle, airplane, bus, train, truck, boat, traffic light (and 70 more)
subsets: val2017
  'val2017':
    length: 184
    categories: label
      label:
        count: 80
        labels: person, bicycle, car, motorcycle, airplane, bus, train, truck, boat, traffic light (and 70 more)
```
And for the command `datum filter source-4 -m a -e '/item/annotation[label="cat" and area > 85]'`, `dinfo` for `source-4` is
```
length: 5000
categories: label
  label:
    count: 80
    labels: person, bicycle, car, motorcycle, airplane, bus, train, truck, boat, traffic light (and 70 more)
subsets: val2017
  'val2017':
    length: 5000
    categories: label
      label:
        count: 80
        labels: person, bicycle, car, motorcycle, airplane, bus, train, truck, boat, traffic light (and 70 more)
```

I imported `source-1` and `source-4` as same dataset, which is coco val2017. And if I applied for `'/annotation[label="cat" and area > 85]'` as filter, It does not work out.

For this command explanation, this part is included in `filter.md` and I linked this page in intermediate skill page for filter. In intermediate skill, we just give simple example to users and if users want more detailed information, they can go to the `filter.md`, check it, and set the mode and filter according to them.